### ISSUE TITLE
Indentation not retrieved from neighboring cpp files

### ISSUE THREAD

**Issue by chetgurevitch** [tpope/vim-sleuth#58]
Vim-sleuth isn't using the formatting from neighboring .cpp or .h files. Detection does work for other filetypes like .rs. I've double checked that the files are in fact being recognized correctly (filetype=cpp) and can reproduce the issue in both neovim (v0.3.1) and a fresh vim install (8.1.333). Installed using vim-plugged, the complete vimrc file used for vim8 is below.

call plug#begin('~/.vim/plugged')
Plug 'tpope/vim-sleuth'
call plug#end()

**Comment by phcerdan** [comment]
I think my issue is related with this one.

`hi.hpp`
```cpp
#include "a.hpp"
#include "b.hpp"
#include "c.hpp"

std::pair<A, B>
remove_edges_and_nodes_from_high_info_graph(const GraphType& g0,
                                            const GraphType& g1,
                                            const double radius = 2.0);

GraphType compare_low_and_high_info_graphs(const GraphType& g0,
                                           const GraphType& g1,
                                           const double radius = 2.0);
```

In that file,  where filetype is `=cpp`
```
:4verbose set shiftwidth?
  shiftwidth=44
        Last set from ~/.vim/plugged/vim-sleuth/plugin/sleuth.vim
```

**Comment by tpope** [comment]
We should probably have a maximum width to be considered valid but I'm not sure what it should be. My Vim help files are detected as 24 due to the hanging indent and I like it that way, even if it's a weird use case.

Regarding the original issue, check `:verbose set shiftwidth?`.

**Comment by sheerun** [comment]
vim-polyglot deals with it by allowing only 2, 4, and 8-spaces indentation, so this file would be ignored when searching for indentation clues

**Comment by tpope** [comment]
Also regarding the original issue, does it work from one `.cpp` file to another? `.h` files can be either C or C++ so I'm not surprised they don't work.

**Comment by tpope** [comment]
Fixed in 3b8cb24af2c985f79f859caadf6016867b5dac7b.
